modules/mariadb/src/test/java/org/testcontainers/containers/MariaDBR2DBCDatabaseContainerTest.java

```diff
@@ -0,0 +1,69 @@
+package org.testcontainers.containers;
+
+import io.r2dbc.spi.Closeable;
+import io.r2dbc.spi.Connection;
+import io.r2dbc.spi.ConnectionFactories;
+import io.r2dbc.spi.ConnectionFactory;
+import io.r2dbc.spi.Result;
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import static org.junit.Assert.*;
+
+public class MariaDBR2DBCDatabaseContainerTest {
```

Comment:
Just wondering if we should prefer an approach similar to our `jdbc-test` module, that runs the same test in a parameterized way?

Reply:
@rnorth wanted to change it to "per module" for better parallelization, see #2520

Reply:
Ah yes, I was assuming something like this actually :grimacing: 
Still, we could maybe extract an abstract base test class or something similar in a later change, to reduced code duplication.

Reply:
Yep, we can split it per-module and have some common base test classes to keep things neat. That's how I'd started with my draft splitting of `jdbc-test`, and works nicely.